content/en/docs/concepts/overview/kubernetes-api.md

```diff
@@ -102,13 +102,21 @@ to ensure that the API presents a clear, consistent view of system resources
 and behavior, and to enable controlling access to end-of-life and/or
 experimental APIs.
 
-Refer to [API versions reference](/docs/reference/using-api/api-overview/#api-versioning)
-for more details on the API version level definitions.
-
 To make it easier to evolve and to extend its API, Kubernetes implements
 [API groups](/docs/reference/using-api/api-overview/#api-groups) that can be
 [enabled or disabled](/docs/reference/using-api/api-overview/#enabling-or-disabling).
 
+API resources are distinguished from each other based on their API groups and
+their resource types, which is referred to as their `kind`s. The API server may
+serve more than one API version for the same resource. All these different
+versions are simply representations of the same resource. Suppose there are two
+versions `v1` and `v1beta1` for the same resource, an object created via the
+`v1beta1` version can then be read, updated, and deleted via the `v1` version.
+The API server handles the conversion between API versions transparently.
```

Comment:
What scope is there for any kind of breaking change (eg adding a mandatory field to an API), and how does Kubernetes handle that kind of situation?

I think this is something like:
- the API server doesn't let you request really old API versions
- there are built in conversions
- where necessary, controllers or other mechanisms might fill in details (maybe at upgrade time)?

Reply:
>  The API server handles the conversion between API versions transparently.

sounds like a strong promise, and I'm worried about the risk of misleading a minority of readers about how good that promise is.

Reply:
I think what I'm looking for here is a paraphrasing of:
> Breaking compatibility of a beta or stable API version, such as v1, is unacceptable. Compatibility for experimental or alpha APIs is not strictly required, but breaking compatibility should not be done lightly, as it disrupts all users of the feature. Alpha and beta API versions may be deprecated and eventually removed wholesale, as described in the deprecation policy.

Once an API has reached beta, you as a developer or cluster operator can rely on being able to read future objects using that beta API version, even if the API reaches GA or even gains a new version after that.
Alpha APIs _probably_ won't break, but you should assume that it might happen and not rely on the opposite.

Reply:
We are talking about different things.

Reply:
> We are talking about different things.

If that's so: maybe we should try to capture that misunderstanding and log an issue about it - if different people think this is about different things, it feels like an area to watch out for.